addons/account/models/account_journal_dashboard.py

```diff
@@ -124,13 +124,28 @@ def get_bar_graph_datas(self):
                 query += " UNION ALL ("+select_sql_clause+" and date >= '"+start_date.strftime(DF)+"' and date < '"+next_date.strftime(DF)+"')"
                 start_date = next_date
 
-        self.env.cr.execute(query, {'journal_id':self.id})
+        self.env.cr.execute(query)
         query_results = self.env.cr.dictfetchall()
         for index in range(0, len(query_results)):
             if query_results[index].get('aggr_date') != None:
                 data[index]['value'] = query_results[index].get('total')
 
         return [{'values': data}]
+        
+        
+    def _get_bar_graph_select_query(self):
+        """
+        Returns the base SELECT SQL query used to gather the bar graph's data.
+        Modifying the data displayed in the graph can be done by overriding
+        this method (but the columns of the result must of course remain the same).
+        Beware that this must be a 'base' query, in the sense the calling function
+        will append UNION expressions to complete it.
+        """
+        
+        return """SELECT sum(residual_company_signed) as total, min(date) as aggr_date 
+               FROM account_invoice 
+               WHERE journal_id = %s and state = 'open'""" % (self.id)
```

Comment:
usually, that way of formatting SQL queries isn't good. It's not your job to format arguments and, if not properly escaped, it may lead to SQL injections possibilities.

Here, there's no chance that self.id could lead to this SQL injection but it's a good habit to have so, please, return a tuple (sql_query, sql_params) and then use the params argument in psycopg execute() method.

```
sql_query, sql_params = self._get_bar_graph_select_query()
self.cr.execute(sql_query, sql_params)
```